simple-facebook-instant-articles.php

```diff
@@ -151,6 +165,38 @@ public function feed_template() {
 	}
 
 	/**
+	 * Set WP query variables for FB IA feed, so we can customise
+	 * what posts are considered for the feed.
+	 *
+	 * @param $query WP_Query object.
+	 */
+	public function customise_feed_query( $query ) {
+
+		$feed_slug = apply_filters( 'simple_fb_feed_slug', $this->token );
+
+		// FB IA feed - customise query.
+		if ( $query->is_main_query() && $query->is_feed( $feed_slug ) ) {
+
+			// Number of posts for feed.
+			$num_posts = intval( apply_filters( 'simple_fb_posts_per_rss', get_option( 'posts_per_rss', 10 ) ) );
+			$query->set( 'posts_per_rss', $num_posts );
+
+			// Exclude sponsored articles.
+			$exclude_sponsored_posts = array(
+				array(
+					'key'     => '_format_sponsored_link',
+					'value'   => '',
+					'compare' => '=',
```

Comment:
@mattheu If I specify a sponsored link, then there is a string stored in DB; Otherwise it's an empty string. So a sponsored post is the one that has got a sponsored link specified coming from `is_sponsored()` in the sponsored module.

I wonder if I need to add another condition - which is meta value doesn't exist?

So overall it's - sponsored link is either empty or doesn't exist.